src/matchPath.js

```diff
@@ -45,6 +45,10 @@ function matchPathBase(end, routePath, urlPath) {
   }
 
   const params = Object.create(null);
+  if (parentParams) {
+    Object.assign(params, parentParams);
```

Comment:
@frenzzy Here child parameters take precedence too, or not? Because if parameter with same name is present in child, it is overwritten by loop below lines you mention